src/libmongoc/tests/unified/result.c

```diff
@@ -565,6 +567,59 @@ result_check (result_t *result,
             goto done;
          }
       }
+
+      if (error_response) {
+         if (!result->reply) {
+            test_set_error (
+               error, "%s", "expected error with a reply, but reply unset");
+            goto done;
+         }
+
+         /* Write operations have the raw response wrapped in a errorReplies
+          * array. Read operations return the raw response as the reply. */
+         bson_iter_t iter;
+         bson_iter_t child;
```

Comment:
Nit: you can potentially move this within the `if (bson_iter_init_find...` condition scope below since `child` isn't referenced outside of that.